packages/containers/src/HeaderBar/HeaderBar.sagas.js

```diff
@@ -0,0 +1,47 @@
+import cmf from '@talend/react-cmf';
+import { call, put, takeEvery, take } from 'redux-saga/effects';
+
+import Connected from './HeaderBar.connect';
+import Constants from './HeaderBar.constant';
+
+/**
+ * This saga takes care of fetching authorized products for the HeaderBar
+ * container according to the provided products URL in the action's payload.
+ * @param {Object} action
+ */
+export function* fetchProducts(action) {
+	const { url, lang } = action.payload;
+	const productsUrl = `${url}?lang=${lang}`;
+
+	yield put(Connected.setStateAction({ fetchingProducts: true }));
+
+	const { response, data } = yield call(cmf.sagas.http.get, productsUrl);
+
+	yield put(Connected.setStateAction({ fetchingProducts: false }));
+
+	if (!response.ok) {
+		return; // Loading products failed, do nothing
```

Comment:
i think the error state should be stated clearly on the component state.
Guessing the status and why it happen is only harder when the state
`not_asked`, `fetching`, `error`, `success data` are not clearly stated